### PR TITLE
Board

### PR DESCRIPTION
**Pull request by dougalg**
This creates a board and related tests. Resolves all tests but the initial placement per user test.

This will definitely need additional logic for non-1-square blocks.

### CODE REVIEW

File: src/js/modules/board.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Board.prototype._sameAdjacentPlayer = function(x, y, playerIndex) {
+    return adjacentSquares(x, y)
+        .some(([_x, _y]) => this._getSquare(_x, _y) === playerIndex);
+}
+
+Board.prototype._getSquare = function(x, y) {
+    return this._squares[x] && this._squares[x][y];
+}
+
+Board.prototype._isStartingPlacment = function (coordinates, playerIndex) {
```
Comment (jonmelcher) [inline src/js/modules/board.js]: typo: Placment -> Placement

File: src/js/modules/board.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    if (this._squares[x] !== undefined && this._squares[x][y] !== undefined) {
+        throw TypeError(`block already placed at [${x}, ${y}]`);
+    }
+    if (this._sameAdjacentPlayer(x, y, playerIndex)) {
+        throw TypeError(`block placed adjacent to same player [${x}, ${y}]`);
+    }
+    if (!this._hasPlayed[playerIndex] && !this._isStartingPlacment(coordinates, playerIndex)) {
+        throw TypeError(`first play by player at invalid location [${x}, ${y}]`);
+    }
+};
```
Comment (jonmelcher) [inline src/js/modules/board.js:41]: we may want a version  of this that returns a bool so we don't need to try/catch during a game

File: src/js/modules/board.js [hunk 3]
```diff
@@ -66,16 +72,11 @@ function adjacentSquares(x, y) {
...
-    }
-    return startingPlacements[playerIndex];
+function adjacentCorners(x, y) {
+    return [
+        [ x - 1, y - 1],
+        [ x + 1, y - 1],
+        [ x - 1, y + 1],
+        [ x + 1, y + 1]
+    ];
 }
```
Comment (jonmelcher) [inline src/js/modules/board.js:82]: woohoo maps

File: test/board_test.js [hunk 4]
```diff
@@ -6,9 +7,9 @@ describe('game board', function() {
...
         it('can be any size', function() {
             let board = new Board(20);
-            strictEqual(board._squares.length, 20);
+            strictEqual(board._size, 20);
             board = new Board(0);
-            strictEqual(board._squares.length, 0);
+            strictEqual(board._size, 0);
         });
         it('excepts on invalid input', function() {
             throws(() => new Board());
```
Comment (jonmelcher) [inline test/board_test.js:15]: LGTM

**Comment by dougalg** [comment]
This will still fail with non 1x1 blocks, woops

**Comment by jonmelcher** [comment]
:8ball:

**Pull request merged**